common/company.py

```diff
@@ -0,0 +1,89 @@
+import datetime
+
+from django.core.exceptions import ValidationError
+from django.core.validators import (
+    MinValueValidator,
+    MaxValueValidator
+)
+from django.db import models
+
+from validation.validate_email import validate_email_custom
+from validation.validate_image import validate_image_file
+
+
+class Company(models.Model):
+    """
+    Abstract base model representing common attributes of companies such as
+    startups and investors.
+
+    Fields:
+    - user: One-to-one link to the associated user account.
+    - industry: Industry category of the company.
+    - company_name: Official name of the company.
+    - location: Geographic location of the company.
+    - logo: Image file for the company's logo, validated for file type and size.
+    - description: Optional textual description of the company.
+    - website: Company's website URL.
+    - email: Contact email address (must be unique).
+    - founded_year: Year the company was founded, constrained between 1900 and current year.
+    - team_size: Number of employees or team members, minimum of 1.
+    - stage: Current stage of the company lifecycle, optional but set by default
+      depending on whether it is a startup or investor.
+    - created_at: Timestamp when the record was created.
+    - updated_at: Timestamp when the record was last updated.
+
+    Validation:
+    - Website must begin with 'http://' or 'https://'.
+    - Founded year cannot be in the future.
+    - Team size must be at least 1.
+    - Description must be at least 10 characters if provided.
+    - Default stage is assigned during validation if not set, based on subclass type.
+
+    This class is abstract and intended to be subclassed.
+    """
+    user = models.OneToOneField(
+        'users.User',
+        on_delete=models.CASCADE
+    )
+    industry = models.ForeignKey(
+        'Industry',
+        on_delete=models.PROTECT
+    )
+    company_name = models.CharField(max_length=254)
+    location = models.ForeignKey(
+        'Location',
+        on_delete=models.PROTECT
+    )
+    logo = models.ImageField(
+        upload_to='company/logos/',
+        validators=[validate_image_file]
+    )
+    description = models.TextField(blank=True, default="")
+    website = models.URLField()
```

Comment:
For consistency, consider making the website field optional with `blank=True`, `default=""` if not all companies will have a website.